File: trunk/quote/Projects/DekalbProperties/Weights.vb
Imports System.ComponentModel
Imports System.Reflection

Imports Model.Quote
Imports Model.Template
Imports Model

Public Class Weights

    Public Shared Function CalcWeight(_Header As Common.Header) As Decimal

        Dim retValue As Decimal = 0
        For Each q As Common.Detail In _Header.Details

            If q.IsWire Then
                Dim p As DisplayableWireProperties = CType(q.QuoteDetailProperties, DisplayableWireProperties)

                Dim w As Decimal = p.PoundsPer1000Feet
                retValue += q.Qty / 3.048 * w / 1000
            End If
        Next

        Return Math.Round(retValue, 4)
    End Function

End Class

File: trunk/quote/Projects/SampleProperties/Weights.cs
using System;

using Model.Quote;
using Model.Template;
using Model;

namespace SampleProperties
{
    class Weights
    {
        public static decimal CalcWeight(Model.Common.Header header)
        {
            decimal retValue = 0;
            foreach(Model.Common.Detail q in header.Details)
            {
                if (q.IsWire)
                {
                    DisplayableWireProperties p 
                        = (DisplayableWireProperties) q.QuoteDetailProperties;

                    decimal w = p.PoundsPer1000Feet;
                    retValue += q.Qty / (decimal)3.048 * w / 1000;
                }
            }

            return Math.Round(retValue, 4);
        }

    }
}
